lib/meataxe.gi

```diff
@@ -3368,41 +3366,37 @@ end;
 ##
 #F  InvariantSesquilinearForm( module ) . . . .
 ##
-## Look for an invariant sesquililinear form of the absolutely irreducible
-## GModule module. Return fail, or the matrix of the form.
+## Look for an invariant sesquililinear form of the GModule module.
+## Return fail, or the matrix of the form.
 SMTX.InvariantSesquilinearForm:=function( module  )
    local DM, q, r, iso, isot, l;
 
-   if not SMTX.IsMTXModule(module) or
-                            not SMTX.IsAbsolutelyIrreducible(module) then
-      Error(
- "Argument of InvariantSesquilinearForm is not an absolutely irreducible module"
-   );
+   if not SMTX.IsMTXModule(module) then
+      Error("Argument of InvariantSesquilinearForm is not a module");
    fi;
-
    if IsBound(module.InvariantSesquilinearForm) then
      return module.InvariantSesquilinearForm;
    fi;
    DM:=SMTX.TwistedDualModule(module);
-   iso:=MTX.IsomorphismIrred(module,DM);
+   iso:=MTX.IsomorphismModules(module,DM);
    if iso = fail then
        SMTX.SetInvariantSesquilinearForm(module, fail);
        return fail;
    fi;
    # Replace iso by a scalar multiple to get iso twisted symmetric
-   q:=Size(module.field);
-   r:=RootInt(q,2);
-   isot:=List( TransposedMat(iso), x -> List(x, y->y^r) );
-   isot:=iso * isot^-1;
-   if not IsDiagonalMat(isot) then
-     Error("Form does not seem to be of the right kind (non-diagonal)!");
-   fi;
-   l:=LogFFE(isot[1,1],Z(q));
-   if l mod (r-1) <> 0 then
-     Error("Form does not seem to be of the right kind (not (q-1)st root)!");
-   fi;
-   iso:=Z(q)^(l/(r-1)) * iso;
-   iso:=ImmutableMatrix(GF(q), iso);
+#    q:=Size(module.field);
+#    r:=RootInt(q,2);
+#    isot:=List( TransposedMat(iso), x -> List(x, y->y^r) );
+#    isot:=iso * isot^-1;
+#    if not IsDiagonalMat(isot) then
+#      Error("Form does not seem to be of the right kind (non-diagonal)!");
+#    fi;
+#    l:=LogFFE(isot[1,1],Z(q));
+#    if l mod (r-1) <> 0 then
+#      Error("Form does not seem to be of the right kind (not (q-1)st root)!");
+#    fi;
+#    iso:=Z(q)^(l/(r-1)) * iso;
```

Comment:
@ThomasBreuer @hulpke @jandebeule any ideas what the commented code above (inside `SMTX.InvariantSesquilinearForm`) is meant to be good for?

It looks to me like an attempt to "normalize" the isomorphism (and hence invariant form) we compute here). But the form can in fact be scaled by an arbitrary scalar (not just by a norm), so I don't understand why it we choose this specific one.

The reason I stumbled over this, and subsequently commented it out (to be deleted) is that it utterly fails if the given module is e.g. the sum of two isomorphic modules - then `isot` above in general won't be diagonal at all (test cases for that added below). On the other hand I can't really think of any reason why it would be useful. But I am sure whoever added this had a good reason, so I wonder what I am missing. 

Any suggestions would be highly welcome!